theano/tensor/basic.py

```diff
@@ -6093,3 +6093,12 @@ def c_code_cache_version(self):
 
     def do_constant_folding(self, node):
         return False
+
+    def connection_pattern(self, node):
+        return [[False] for i in node.inputs]
+
+    def grad(self, inputs, grads):
+        return [DisconnectedType()() for i in inputs]
+
+    def R_op(self, inputs, eval_points):
+        return [zeros(inputs, self.dtype)]
```

Comment:
There is few ops that implement rop. I think it would be quick to check if
there is other op that don't implement it correctly. Would it be useful?

On Fri, Oct 16, 2015 at 2:19 PM, Pascal Lamblin <notifications@github.com>
wrote:

> In theano/tensor/basic.py
> <https://github.com/Theano/Theano/pull/3521#discussion_r42273371>:
>
> > @@ -6093,3 +6093,12 @@ def c_code_cache_version(self):
> >
> >      def do_constant_folding(self, node):
> >          return False
> > +
> > +    def connection_pattern(self, node):
> > +        return [[False] for i in node.inputs]
> > +
> > +    def grad(self, inputs, grads):
> > +        return [DisconnectedType()() for i in inputs]
> > +
> > +    def R_op(self, inputs, eval_points):
> > +        return [zeros(inputs, self.dtype)]
>
> This is probably because IncSubtensor.R_op returns None if *either*
> differentiable input is None, whereas it should only do so if *both* are.
> Since there may be other Ops doing the same error, I think we should be on
> the side of correct results, rather than maximum speed. So I agree with
> your current implementation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3521/files#r42273371>.
>